gslib/tests/testcase/base.py

```diff
@@ -194,9 +195,10 @@ def CreateTempFile(self, tmpdir=None, contents=None, file_name=None,
     Returns:
       The path to the new temporary file.
     """
-    tmpdir = tmpdir or self.CreateTempDir()
-    file_name = file_name or self.MakeTempName('file')
-    if isinstance(file_name, six.text_type):
+
+    tmpdir = six.ensure_str(tmpdir or self.CreateTempDir())
+    file_name = file_name or self.MakeTempName(str('file'))
```

Comment:
Should we do `six.ensure_string('file')` here instead of `str('file')`? Since `str` can return either bytes or unicode depending on version, and a string literal should also imply`str`.

Reply:
ensure_string should return the same as str(). The reason for using str instead of text/binary is that join() requires str() type and does not work with both text/binary types in all version of python.

Reply:
Got it. Is it necessary to wrap with `str()` at all then? Since an inline string literal should evaluate to the interpreter's default string type.

Reply:
I understand that there will be different encodings when declared with str() and I try to avoid it when possible, but I believe it is the more universal situation in this instance.

Reply:
The default is overridden with the top line `# -*- coding: utf-8 -*-` and can cause join to not work properly (at least that is the behavior I observed during my testing)

Reply:
Right, but `'foo' == str('foo')` is what I was meaning. The `str()` would be redundant. :slightly_smiling_face:

Reply:
Oooh I see now, re: the utf-8 string at the top